src/playback-watcher.js

```diff
@@ -377,27 +345,45 @@ export default class PlaybackWatcher {
 
     const sourceUpdater = this.masterPlaylistController_.sourceUpdater_;
     const buffered = this.tech_.buffered();
+    const audioBuffered = sourceUpdater.audioBuffer ? sourceUpdater.audioBuffered() : null;
+    const videoBuffered = sourceUpdater.videoBuffer ? sourceUpdater.videoBuffered() : null;
+
+    // verify that at least two segment durations have been
+    // appended before checking for a gap.
+    const twoSegmentDurations = (this.media().targetDuration - Ranges.TIME_FUDGE_FACTOR) * 2;
+    const bufferedToCheck = [audioBuffered, videoBuffered];
+
+    for (let i = 0; i < bufferedToCheck.length; i++) {
+      // skip null buffered
+      if (!bufferedToCheck[i]) {
+        continue;
+      }
 
-    if (
-      closeToBufferedContent({
-        buffered,
-        audioBuffered: sourceUpdater.audioBuffer ? sourceUpdater.audioBuffered() : null,
-        videoBuffered: sourceUpdater.videoBuffer ? sourceUpdater.videoBuffered() : null,
-        targetDuration: this.media().targetDuration,
-        currentTime
-      })
-    ) {
-      const nextRange = Ranges.findNextRange(buffered, currentTime);
+      const timeAhead = Ranges.timeAheadOf(bufferedToCheck[i], currentTime);
 
-      seekTo = nextRange.start(0) + Ranges.SAFE_TIME_DELTA;
-      this.logger_(`Buffered region starts (${nextRange.start(0)}) ` +
-                   ` just beyond seek point (${currentTime}). Seeking to ${seekTo}.`);
+      // if we are less than two video/audio segment durations behind,
+      // we haven't appended enough to call this a bad seek.
+      if (timeAhead < twoSegmentDurations) {
+        return false;
+      }
+    }
 
-      this.tech_.setCurrentTime(seekTo);
-      return true;
+    const nextRange = Ranges.findNextRange(buffered, currentTime);
+
+    // we have appended enough content, but we don't have anything buffered
+    // to seek over the gap
+    if (nextRange.length === 0) {
+      return false;
     }
```

Comment:
Any specific reason for moving all of this out of a separate function?

Reply:
After reducing the function to just the `twoSegmentDurations` check and the `nextRange` check it seemed silly to  call `findNextRange` in the helper function and then directly afterwards since we need access to it. So I think it made more sense to move the helper function in here.